### DISCUSSION TITLE
I want to use DataTrigger.

### DISCUSSION THREAD

**Discussion by dimohy** [unoplatform/uno#11530] [Q&A] [answered]
DataTrigger is occasionally useful to me. However, I can't figure out how to use DataTrigger on the Uno Platform.

How can I reference DataTrigger and use it in XAML?

**Comment by jeromelaban** [comment, marked as answer]
You can find the `DataTrigger` in the UWP/WinUI Behaviors SDK. See [this post for details](https://stackoverflow.com/questions/31929071/trigger-element-xaml-is-not-supported-in-a-uwp-project).

Uno has its own port of the behaviors that [are available here](https://github.com/unoplatform/Uno.XamlBehaviors).

**Reply by dimohy** [reply]
Thank you for your response.

I was expecting something like DataTrigger in the MAUI.

```csharp
<Entry x:Name="entry"
       Text=""
       Placeholder="Enter text" />
<Button Text="Save">
    <Button.Triggers>
        <DataTrigger TargetType="Button"
                     Binding="{Binding Source={x:Reference entry},
                                       Path=Text.Length}"
                     Value="0">
            <Setter Property="IsEnabled"
                    Value="False" />
            <!-- Multiple Setter elements are allowed -->
        </DataTrigger>
    </Button.Triggers>
</Button>
```

Since the Uno structure follows the WinUI, it was to be expected that it would not support DataTrigger natively.

However, it could have supported data triggers like the MAUI, which is not a behavior.

Maybe I just didn't find it?

**Reply by jeromelaban** [reply]
Indeed, `DataTrigger` is not a feature supported directly by WinUI, and only from the UWP/WinUI Xaml [DataTriggerBehavior](https://github.com/Microsoft/XamlBehaviors/wiki/DataTriggerBehavior). That being said, you can implement the same feature by using [x:Bind](https://platform.uno/docs/articles/features/windows-ui-xaml-xbind.html?q=bind) functions (e.g. `<Button IsEnabled="{x:Bind local:HasStringLengthNotZero(entry.Text)}" />` given a `HasStringLengthNotZero` method you can add yourself).

**Comment by ACheekyBeluga** [comment]
This package doesn't appear to work in the latest versions of Uno

**Reply by jeromelaban** [reply]
Can you give more details? The packages are compatible with the latest versions of Uno.

**Reply by ACheekyBeluga** [reply]
My mistake, I wasn't including it properly and had tried to import the Uno package across all platforms. Following the Uno package documentation, this works fine using conditional inclusion : 
```xaml
<ItemGroup Condition="'$(TargetFramework)' == 'net8.0-windows10.0.19041'">
	<PackageReference Include="Microsoft.Xaml.Behaviors.WinUI.Managed" Version="2.0.9" />
</ItemGroup>
<ItemGroup Condition="'$(TargetFramework)' != 'net8.0-windows10.0.19041'">
	<PackageReference Include="Uno.Microsoft.Xaml.Behaviors.WinUI.Managed" Version="2.4.2" />
</ItemGroup>
```
